samples/nrf9160/aws_fota/src/main.c

```diff
@@ -0,0 +1,571 @@
+	int at_socket_fd;
+	int bytes_written;
+	int bytes_read;
+	char imei_buf[IMEI_LEN + 1];
+	int ret;
+
+	at_socket_fd = nrf_socket(NRF_AF_LTE, 0, NRF_PROTO_AT);
+	__ASSERT_NO_MSG(at_socket_fd >= 0);
+
+	bytes_written = nrf_write(at_socket_fd, "AT+CGSN", 7);
+	__ASSERT_NO_MSG(bytes_written == 7);
+
+	bytes_read = nrf_read(at_socket_fd, imei_buf, IMEI_LEN);
+	__ASSERT_NO_MSG(bytes_read == IMEI_LEN);
+	imei_buf[IMEI_LEN] = 0;
+
+	snprintf(id, CLIENT_ID_LEN + 1, "nrf-%s", imei_buf);
+
+	ret = nrf_close(at_socket_fd);
+	__ASSERT_NO_MSG(ret == 0);
+#else
+#error Missing NRF_CLOUD_CLIENT_ID
+#endif /* defined(CONFIG_BSD_LIBRARY) */
+#else
+	memcpy(id, CONFIG_CLOUD_CLIENT_ID, CLOUD_CLIENT_ID_LEN + 1);
+#endif /* !defined(NRF_CLOUD_CLIENT_ID) */
+
+	return 0;
+}
+
+/**@brief Initialize the MQTT client structure
+*/
+static int client_init(struct mqtt_client *client)
+{
+	mqtt_client_init(client);
+	provision();
+	broker_init();
+
+	int ret = client_id_get(client_id_buf);
+	if (ret != 0) {
+		return ret;
+	}
+
+	/* MQTT client configuration */
+	client->broker = &broker;
+	client->evt_cb = mqtt_evt_handler;
+	client->client_id.utf8 = client_id_buf;
+	client->client_id.size = CLIENT_ID_LEN;
+	client->password = NULL;
+	client->user_name = NULL;
+	client->protocol_version = MQTT_VERSION_3_1_1;
+
+	/* MQTT buffers configuration */
+	client->rx_buf = rx_buffer;
+	client->rx_buf_size = sizeof(rx_buffer);
+	client->tx_buf = tx_buffer;
+	client->tx_buf_size = sizeof(tx_buffer);
+
+	/* MQTT transport configuration */
+	client->transport.type = MQTT_TRANSPORT_SECURE;
+
+	static sec_tag_t sec_tag_list[] = {CONFIG_NRF_CLOUD_SEC_TAG};
+	struct mqtt_sec_config *tls_config = &(client->transport).tls.config;
+
+	tls_config->peer_verify = 2;
+	tls_config->cipher_list = NULL;
+	tls_config->cipher_count = 0;
+	tls_config->sec_tag_count = ARRAY_SIZE(sec_tag_list);
+	tls_config->sec_tag_list = sec_tag_list;
+	tls_config->hostname = CONFIG_AWS_HOSTNAME;
+
+	return 0;
+
+}
+
+/**@brief Initialize the file descriptor structure used by poll.
+*/
+static int fds_init(struct mqtt_client *c)
+{
+	if (c->transport.type == MQTT_TRANSPORT_NON_SECURE) {
+		fds.fd = c->transport.tcp.sock;
+	} else {
+#if defined(CONFIG_MQTT_LIB_TLS)
+		fds.fd = c->transport.tls.sock;
+#else
+		return -ENOTSUP;
+#endif
+	}
+
+	fds.events = POLLIN;
+
+	return 0;
+}
+
+/**@brief Configures modem to provide LTE link. Blocks until link is
+ * successfully established.
+ */
+static void modem_configure(void)
+{
+#if defined(CONFIG_LTE_LINK_CONTROL)
+	if (IS_ENABLED(CONFIG_LTE_AUTO_INIT_AND_CONNECT)) {
+		/* Do nothing, modem is already turned on
+		 * and connected.
+		 */
+	} else {
+		int err;
+
+		printk("LTE Link Connecting ...\n");
+		err = lte_lc_init_and_connect();
+		__ASSERT(err == 0, "LTE link could not be established.");
+		printk("LTE Link Connected!\n");
+	}
+#endif
+}
+
+void aws_fota_cb_handler(enum aws_fota_evt_id evt)
+{
+	/* When the applicaiton is ready restart */
```

Comment:
This comment looks wrong.

Reply:
Should be "Notify application that FOTA is done and the application should restart/the library is ready for the application to restart.